common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java

```diff
@@ -233,7 +234,12 @@ public boolean mouseClicked(double mouseX, double mouseY, int button) {
             if (hovered != null && !(hovered instanceof TerritoryPoi)) {
                 McUtils.playSound(SoundEvents.EXPERIENCE_ORB_PICKUP);
                 if (hovered.hasStaticLocation()) {
-                    CompassModel.setCompassLocation(new Location(hovered.getLocation()));
+                    if (hovered instanceof IconPoi) {
```

Comment:
```suggestion
                    if (hovered instanceof IconPoi iconPoi) {
```

This way, you don't need to cast again.